nomad/node_endpoint.go

```diff
@@ -261,61 +261,75 @@ func (n *Node) Deregister(args *structs.NodeDeregisterRequest, reply *structs.No
 	}
 
 	// Verify the arguments
-	if args.NodeID == "" {
-		return fmt.Errorf("missing node ID for client deregistration")
+	if len(args.NodeIDs) == 0 {
```

Comment:
This currently does not handle receiving a de-register request from an older server - until we remove the field, we still need to handle receiving an individual `NodeID`